src/test/java/org/broadinstitute/consent/http/resources/DatasetResourceTest.java

```diff
@@ -95,47 +108,78 @@ public void testCreateDataset() throws Exception {
+    public void testCreateDatasetNoJson() {
+        initResource();
+        Response response = resource.createDataset(authUser, uriInfo, "");
+        assertEquals(400, response.getStatus());
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void testCreateDatasetNoProperties() {
+        initResource();
+        Response response = resource.createDataset(authUser, uriInfo, "{\"properties\":[]}");
+        assertEquals(400, response.getStatus());
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void testCreateDatasetMissingName() {
+        String json = createPropertiesJson("Dataset Name", null);
+        // TODO: Rewrite createDataset to handle empty name strings and entirely missing name properties
 
         initResource();
-        Response responseNoJson = resource.createDataset(authUser, uriInfo, "");
-        assertEquals(400, responseNoJson.getStatus());
+        Response response = resource.createDataset(authUser, uriInfo, json);
+        assertEquals(400, response.getStatus());
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void testCreateDatasetInvalidProperty() {
+        List<DataSetPropertyDTO> invalidProperties = new ArrayList<>();
+        invalidProperties.add(new DataSetPropertyDTO("Invalid Property", "test"));
+        when(datasetService.findInvalidProperties(any())).thenReturn(invalidProperties);
+
+        String json = createPropertiesJson(invalidProperties);
+
+        initResource();
+        Response response = resource.createDataset(authUser, uriInfo, json);
+        assertEquals(400, response.getStatus());
+    }
 
-        Response responseNoProperties = resource.createDataset(authUser, uriInfo, "{\"properties\":[]}");
-        assertEquals(400, responseNoProperties.getStatus());
+    @Test(expected = BadRequestException.class)
+    public void testCreateDatasetDuplicateProperties() {
+        List<DataSetPropertyDTO> duplicateProperties = new ArrayList<>();
+        duplicateProperties.add(new DataSetPropertyDTO("Dataset Name", "test"));
+        duplicateProperties.add(new DataSetPropertyDTO("Dataset Name", "test"));
+        when(datasetService.findDuplicateProperties(any())).thenReturn(duplicateProperties);
 
-        Response responseMissingDatasetName = resource.createDataset(authUser, uriInfo,
-              "{\"properties\":[{\"propertyName\":\"Species\",\"propertyValue\":\"test\"}]}");
-        assertEquals(400, responseMissingDatasetName.getStatus());
+        String json = createPropertiesJson(duplicateProperties);
 
-        Response responseInvalidProperty = resource.createDataset(authUser, uriInfo,
-              "{\"properties\":[{\"propertyName\":\"Invalid Property\",\"propertyValue\":\"test\"}]");
-        assertEquals(400, responseInvalidProperty.getStatus());
+        initResource();
+        Response response = resource.createDataset(authUser, uriInfo, json);
+        assertEquals(400, response.getStatus());
```

Comment:
This test never reaches this line since line 163, `resource.createDataset`, throws the expected exception.